apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solrj/SolrClientInterceptor.java

```diff
@@ -0,0 +1,194 @@
+                            span.tag(SolrjTags.TAG_DELETE_TYPE, "deleteByQuery");
+                            span.tag(SolrjTags.TAG_DELETE_VALUE, deleteQuery.toString());
+                        }
+                    } else {
+                        span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+                        span.tag(SolrjTags.TAG_DOCS_SIZE, String.valueOf(documents.size()));
+                        span.tag(SolrjTags.TAG_COMMIT_WITHIN, String.valueOf(update.getCommitWithin()));
+                    }
+                }
+            } else {
+                actionName = action.name();
+                if (action == AbstractUpdateRequest.ACTION.COMMIT) {
+                    span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+
+                    span.tag(SolrjTags.TAG_COMMIT, "true");
+                    span.tag(SolrjTags.TAG_SOFT_COMMIT, params.get(UpdateParams.SOFT_COMMIT, ""));
+                } else {
+                    span = getSpan(collection, request.getPath(), actionName, instance.getRemotePeer());
+
+                    span.tag(SolrjTags.TAG_OPTIMIZE, "true");
+                    span.tag(SolrjTags.TAG_MAX_OPTIMIZE_SEGMENTS, params.get(UpdateParams.MAX_OPTIMIZE_SEGMENTS, "1"));
+                }
+            }
+            span.tag(SolrjTags.TAG_ACTION, actionName);
+        } else if (request instanceof QueryRequest) { // It failed to spot that through 'path' when it had defined QueryType.
+            String operatorName = String.format("solrJ/%s%s", collection, request.getPath());
+            span = ContextManager.createExitSpan(operatorName, instance.getRemotePeer())
+                    .setComponent(ComponentsDefine.SOLRJ)
+                    .setLayer(SpanLayer.DB);
+        } else { // No admin API
+            return;
+        }
+
+        span.tag(SolrjTags.TAG_PATH, request.getPath());
+        span.tag(SolrjTags.TAG_COLLECTION, collection);
+        span.tag(SolrjTags.TAG_METHOD, request.getMethod().name());
+        span.tag(SolrjTags.TAG_QT, params.get(CommonParams.QT, request.getPath()));
+
+        Context.get().start();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+                              Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        if (!ContextManager.isActive()) return ret;
+
+        final Context context = Context.remove();
+        final long elapse = System.currentTimeMillis() - context.getStartTime();
+
+        AbstractSpan span = ContextManager.activeSpan();
+        NamedList<Object> response = (NamedList<Object>) ret;
+        if (response != null) {
+            NamedList<Object> header = (NamedList<Object>) response.get("responseHeader");
+            if (header != null) { // common
+                span.tag(SolrjTags.TAG_STATUS, header.get("status").toString());
```

Comment:
could u confirm `header.get("status")` isn't null ?

Reply:
Yes, it cannot be null. Just like status code in http response.

Reply:
Thanks for your suggestion.
I change to String.valueOf(...).